### DISCUSSION TITLE
What is the purpose of running `cargo check` in `buildDepsOnly`?

### DISCUSSION THREAD

**Discussion by dpc** [ipetkov/crane#86] [Q&A] [answered]
I thought it would only be useful if someone wanted to run `cargo check` in other packages, but 99% the time crane-based setups will actually want to do a full compilation with `cargo build` anyway, no?

**Comment by dpc** [comment]
BTW. Documentation for `cargoCheckCommand` in `buildDepsOnly` incorrectly states that the default is `cargo build --profile release`, while it really seems to be `cargoWithProfile check --all-targets` so `s/build/check/g`

**Reply by ipetkov** [reply]
Good eye, totally missed this

**Comment by ipetkov** [comment, marked as answer]
`buildDepsOnly` runs `cargo check` so that it can cache all the dependency artifacts needed to run `cargo clippy` (or `cargo check`) on the crate itself without having to repeat that work each time. 

This includes building and linking proc-macros among other metadata used for type checking

**Reply by dpc** [reply]
Oh, that's good to know.
